docs/Contributors_Guide/directory_tree.rst

```diff
@@ -0,0 +1,74 @@
+**************
+Directory Tree
+**************
+
+Goals
+=====
+* No conflicts between different packages in their external packages
+* No circular dependencies
+* If a user/dev introduces a new dependency they need to add an entry in environment.yaml and setup.cfg 
+
+Action Items
+============
+Need GitHub Actions that test the setup.cfg and environment.yml to ensure that our code runs within
+Python and follows this type of generic structure:
+
+Python structure::
+
+   pkg
+   ├── __init__.py
+   ├── module1.py
+   └── subpkg
+     ├── __init__.py
+     └── module2.py
+      
+What are modules and subpkg?
+
+* Tests follow similar structure as code.
+
+* Workflow tools are standalone pieces of code that can be used with other tools
+  helpers can be utilized by any tool/code at any level.
+  
+* New tools should be added as subpkgs of uwtools.
+
+* New helpers should be added as modules of helpers
+
+
+Sample Structure::
+
+  tests
+    test_utils
+      __init__.py
+      test_logger.py
+      test_errors.py
+    test_scheduler
+      __init__.py
+      test_slurm.py
+    test_runners
+      __init__.py
+      test_forecast.py
```

Comment:
What are runners?
Seems like out of scope of the workflow-tools that call out concrete use of a generic task.

Reply:
@christinaholtNOAA @fgabelmannjr  ?

Reply:
I think that this and the other files are meant to be illustrative, not prescriptive.

Reply:
I concur with @benjamin-cash .  My understanding of this task was to provide an illustrative example to contributors for how contributions should be organized within the repository rather than a prescriptive list of the structure.   However, as the framework matures, I could see this documentation becoming more prescriptive.

Reply:
I've made some edits to try and bring the illustrative/sample nature out a bit more.